amads/melody/mel_accent.py

```diff
@@ -0,0 +1,129 @@
+__author__ = "Yiwen Zhao"
+
+import numpy as np
+from ..core.basics import Score, Note, Part
+
+def calculate_melodic_accent(score: Score) -> list:
+    """
+    Calculate melodic accent salience according to Thomassen's model.
+    
+    Melodic accent is determined by analyzing patterns of three consecutive notes.
+    The model assigns different accent values based on the following patterns:
+    - Peak accent (up-down pattern): 1.0
+    - Valley accent (down-up pattern): 0.6
+    - Continuous ascending: 0.4
+    - Continuous descending: 0.4
+    - Other patterns (including repeated notes): 0.0
+    
+    Parameters
+    ----------
+    score : Score
+        A Score object containing the melody to analyze. The score will be
+        flattened and collapsed into a single sequence of notes ordered by
+        onset time.
+    
+    Returns
+    -------
+    list
+        A list of accent values between 0 and 1 for each note. First and last
+        notes always have accent value 0 as they require three notes for
+        accent calculation.
+    
+    Examples
+    --------
+    >>> score = Score()
+    >>> part = Part()
+    >>> notes = [Note(pitch=60), Note(pitch=64), Note(pitch=62)]  # up-down pattern
+    >>> for note in notes:
+    ...     part.append(note)
+    >>> score.append(part)
+    >>> accent_values = calculate_melodic_accent(score)
+    >>> print(accent_values)
+    [0.0, 1.0, 0.0]  # Peak accent in the middle
+    """
+    # Flatten and collapse the score into a single sequence of notes
+    flattened_score = score.flatten(collapse=True)
```

Comment:
I think we should check if the score is monophonic and throw an error if not

Reply:
If monophonic means strictly non-overlapping, then almost any melody recorded on a MIDI keyboard is likely to fail the monophonic test due to slightly overlapping notes.  It might still be best to force users to explicitly convert to monophonic non-overlapping -- at least they won't be surprised when a clearly chordal performance sneaks in and gives confusing results.

Reply:
Hmm, that's a good point. I suppose many of these algorithms will work as intended with such a melody (i.e. one with slight overlaps)? In such cases, maybe my `is_monophonic` test would do more harm than good. Maybe it's the user's fault if they run it on something stupid...

Reply:
We could offer a `make_monophonic` function to map from "sort of" to "really" monophonic MIDI? 

Clear principles and documentation for the design needed.

E.g., each new note start takes precedence, cutting off everything else.

Then for any exactly coincident events? That would be one of the "user's fault" cases Peter mentions ;). Perhaps raising an error in that case is best, to draw a line under how far we're prepared to accommodate inappropriate scores (exact coincidence indicating a fully symbolic score in multiple parts).